src/ayb_db/db_interfaces.rs

```diff
@@ -191,7 +191,14 @@ WHERE
                 .bind(entity_slug)
                 .bind(database_slug)
                 .fetch_one(&self.pool)
-                .await?;
+                .await
+                .or_else(|err| match err {
+                    sqlx::Error::RowNotFound => Err(AybError::RecordNotFound {
```

Comment:
I discovered a tiny bug when I added a test for retrieving databases with a case-sensitive entity slug. Adding it to this PR since the test discovered the issue :).